src/crds_gossip_push.rs

```diff
@@ -177,6 +178,11 @@ impl CrdsGossipPush {
             if new_items.get(&val.0.pubkey()).is_some() {
                 continue;
             }
+            if let Some(contact) = val.1.value.contact_info() {
+                if !ClusterInfo::is_valid_address(&contact.ncp) {
```

Comment:
the functionality for valid IPs can be closer to network code to avoid a dependency from crds to cluster_info...

Reply:
Okay, will do.

Reply:
@sagar-solana i would stick it in ContactInfo or something

Reply:
Oh it does? alright, I'll refactor.